msbuild/Xamarin.Mac.Tasks/Tasks/CompileAppManifest.cs

```diff
@@ -38,12 +39,18 @@ public override bool Execute ()
 				return false;
 			}
 
+			if (AssemblyName.IndexOf ('.') != AssemblyName.LastIndexOf ('.'))
+				LogAppManifestError ("The assembly name ({0}) cannot contain more than one dot.", AssemblyName);
```

Comment:
Looks like this was part of the code I copied over from iOS (when I first looked at extensions last may).  I'll update it to check for the .app instead like iOS is doing now